packages/blade/src/components/Input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,227 @@
+import React, { useState } from 'react';
+import type { ReactElement, ReactNode } from 'react';
+import type { TextInput as TextInputReactNative } from 'react-native';
+import type { BaseInputProps } from '~components/Input/BaseInput';
+import { BaseInput } from '~components/Input/BaseInput';
+import type { IconComponent } from '~components/Icons';
+import { InfoIcon, CloseIcon } from '~components/Icons';
+import { IconButton } from '~components/Button/IconButton';
+import { getPlatformType, isEmpty } from '~utils';
+import { CharacterCounter } from '~components/Form/CharacterCounter';
+import Box from '~components/Box';
+
+export type TextInputProps = Pick<
+  BaseInputProps,
+  | 'label'
+  | 'labelPosition'
+  | 'necessityIndicator'
+  | 'validationState'
+  | 'helpText'
+  | 'errorText'
+  | 'successText'
+  | 'placeholder'
+  | 'type'
+  | 'defaultValue'
+  | 'name'
+  | 'onChange'
+  | 'onBlur'
+  | 'value'
```

Comment:
I guess it's time now we create reusable types for FormLabel, FormHint etc.

Reply:
i've already created under `Form/FormTypes.ts`

Reply:
Ouuu nice didn't noticed. 
Then in omit lets do 

```
  | keyof FormInputLabelProps
  | keyof FormInputValidationProps
```

Reply:
didn't get you? which omit you're referring to?

Reply:
Oh sorry not omit. Pick. 

```ts
export type TextInputProps = Pick<
  BaseInputProps,
  | keyof FormInputLabelProps
  | keyof FormInputValidationProps
```

Reply:
keeping it as is since too much abstraction otherwise to jump and see in favor of readability.